### PR TITLE
Deng 850 firefox android clients reported ping

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
Implements [DENG-850](https://mozilla-hub.atlassian.net/browse/DENG-850).

The specific test scenarios are defined in [DENG-884](https://mozilla-hub.atlassian.net/browse/DENG-884).

[DENG-850]: https://mozilla-hub.atlassian.net/browse/DENG-850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DENG-884]: https://mozilla-hub.atlassian.net/browse/DENG-884?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_baseline_no_first_session/fenix.first_session.schema.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+- client_info:
+    client_id: client-2
```
Comment (fbertsch) [inline tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_baseline_no_first_session/fenix.first_session.schema.yaml:3]: If I'm reading this right, a client who sends just a `first_session` ping won't have an entry in this table? I'm not sure if that's the desired behavior

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_first_session_no_baseline/fenix.baseline_clients_first_seen.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+- client_id: client-5
```
Comment (fbertsch) [inline tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_first_session_no_baseline/fenix.baseline_clients_first_seen.yaml:2]: Why is `client-5` not included in `expect.yaml` with both `reported_first_session_ping = False` and `reported_metrics_ping = False`?

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/fenix.baseline_clients_first_seen.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+- client_id: client-1
```
Comment (fbertsch) [inline tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/fenix.baseline_clients_first_seen.yaml:2]: same here for `client-1`

File: sql/moz-fx-data-shared-prod/telemetry_derived/event_types_history_v1/metadata.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ labels:
   incremental: true
   schedule: daily
 # see https://bugzilla.mozilla.org/show_bug.cgi?id=1805722#c10
-#scheduling:
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-shared-prod/telemetry_derived/event_types_history_v1/metadata.yaml:19]: Clean-up of YAML warnings.

**Review by fbertsch — COMMENTED** [review]
@lucia-vargas-a I don't see any changes to `query.sql` - does that mean there wasn't a bug?

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_first_session_no_baseline/expect.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
```
Comment (lucia-vargas-a) [inline tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_first_session_no_baseline/expect.yaml:1]: client-2 reported baseline ping.
client-1 and client-3 didn't report a baseline ping but still do appear with their corresponding values for reported metrics and first_session ping.

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/expect.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (lucia-vargas-a) [inline tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/expect.yaml:1]: client-3 doesn't report a baseline nor a metrics ping in this submission_date, but it did report a metrics ping at some point, as seen in _previous (existing table records).
client-2 did not report a baseline ping and doesn't have a record in _previous. It is only reporting a metrics ping.

**Review by fbertsch — APPROVED** [review]
LGTM. 1 question about sample_id, just to make sure when we join with this table on `(client_id, sample_id)` we will get the correct results

File: tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/expect.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  activated: false
+  adjust_network: 'Organic'
+  channel: release
+  metadata:
+    reported_first_session_ping: false
+    reported_metrics_ping: true
+    min_metrics_ping_submission_date: 2023-01-01
+    adjust_network__source_ping_datetime: '2023-01-01T00:00:00'
+    adjust_network__source_ping: metrics
+- client_id: client-2
```
[the remark this answers is not in the document]
Comment (lucia-vargas-a) [reply tests/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/test_only_metrics_ping/expect.yaml:16]: It should if it's either in the baseline or in the metrics ping -but it wasn't-.   I added it though to validate this case too.

**Pull request merged**